frontend/src/components/LoginDialog.jsx

```diff
@@ -22,19 +30,56 @@ export default function LoginDialog() {
   };
 
   const handleClose = () => {
+    setOpen(false);
+    setRegisterVisible('none');
+    setErrorAlertVisible('none');
+  };
+
+  const handleCloseLogin = () => {
     authService.getToken(user, password).then((data) => 
       cookies.set('token', data.token, {httpOnly: false})
     );
     setOpen(false);
+    setRegisterVisible('none');
+    setErrorAlertVisible('none');
+  };
+
+  const handleCloseRegister = () => {
+    if(email.includes('@')){
+        authService.register(user, password, password2, email).then((data) => 
+          console.log(data)
+          // cookies.set('token', data.token, {httpOnly: false})
+        );
+        setOpen(false);
+        setRegisterVisible('none');
+        setErrorAlertVisible('none');
+        // TODO: nice alert
+        alert("Account has been created!.")
+    }
+    else {
+      setErrorAlertVisible('block')
+    }
+    
   };
 
+  const expandRegisterForm = () => {
+    setRegisterVisible('block')
+  }
+  
+
   return (
     <div>
-      <Button variant="outlined" onClick={handleClickOpen}>
+      
+      <Button variant="outlined" onClick={handleClickOpen} style={{backgroundColor: "#2b2b69", color: "white"}}>
         Login
       </Button>
       <Dialog open={open} onClose={handleClose}>
         <DialogTitle>Login</DialogTitle>
+        <Alert severity="error" style={{display: errorAlertVisible}}>
```

Comment:
There's a better way to handle validation

https://mui.com/material-ui/react-text-field/#validation